File: bcode/src/main/java/edu/abani/bcode/annot/BCode.java
package edu.abani.bcode.annot;


import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target(TYPE)
@Retention(RUNTIME)
public @interface BCode {
    String table() default "user_details";
    String column() default "id";
    String prefix() default "SYS";

}

File: seqgen/src/main/kotlin/edu/abani/seqgen/annotation/BCode.kt
package edu.abani.seqgen.annotation

import java.lang.annotation.ElementType
import java.lang.annotation.RetentionPolicy

@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class BCode(val table: String = "user_detail",
                       val column: String = "id")
